In [6]:
import pandas as pd

# Load financial data
df = pd.read_csv("data.csv")

print("Financial Data Chatbot")
print("Available companies: Microsoft, Tesla, Apple")
print("Available years: 2023, 2024, 2025")

# Get user input
company = input("Enter company name: ").strip()
year = int(input("Enter fiscal year (2023-2025): "))

# Filter the data
result = df[
    (df["Company"].str.lower() == company.lower()) &
    (df["Fiscal Year"] == year)
]

# Display result
if not result.empty:
    row = result.iloc[0]

    print("\nFinancial Information")
    print("----------------------")
    print("Company:", row["Company"])
    print("Fiscal Year:", row["Fiscal Year"])
    print("Total Revenue ($M):", row["Total Revenue ($M)"])
    print("Net Income ($M):", row["Net Income ($M)"])
    print("Total Assets ($M):", row["Total Assets ($M)"])
    print("Total Liabilities ($M):", row["Total Liabilities ($M)"])
    print("Operating Cash Flow ($M):", row["Operating Cash Flow ($M)"])

else:
    print("\nSorry, no data found for that company and year.")

Financial Data Chatbot
Available companies: Microsoft, Tesla, Apple
Available years: 2023, 2024, 2025

Financial Information
----------------------
Company: Microsoft
Fiscal Year: 2023
Total Revenue ($M): 211915
Net Income ($M): 72361
Total Assets ($M): 411976
Total Liabilities ($M): 205753
Operating Cash Flow ($M): 87582


FINANCIAL CHATBOT DOCUMENTATION

1. Purpose
This chatbot provides financial information for Microsoft, Tesla,
and Apple using data stored in a CSV file.

2. How It Works
The chatbot uses Python and Pandas to read the financial dataset.
The user enters a company name and fiscal year. Pandas then filters
the dataset and displays the corresponding total revenue.

3. User Inputs
- Company: Microsoft, Tesla, or Apple
- Fiscal Year: 2023, 2024, or 2025

4. Example
Input:
Company: Microsoft
Fiscal Year: 2025

Output:
Microsoft revenue in 2025: $281,724 million

5. Testing
The chatbot was tested with different companies and fiscal years.
The returned values were checked against the original dataset.

6. Limitations
- The chatbot only supports Microsoft, Tesla, and Apple.
- It only contains data for fiscal years 2023, 2024, and 2025.
- It currently provides total revenue only.
- It cannot answer questions outside the available dataset.
- Company names and fiscal years must be entered in a valid format.

7. Technologies Used
- Python
- Pandas
- CSV dataset

## Test Results

The chatbot was tested with different companies and fiscal years.

| Company | Fiscal Year | Expected Revenue ($M) | Chatbot Output ($M) | Status |
|---------|-------------|------------------------|----------------------|--------|
| Microsoft | 2023 | 211,915 | 211,915 | PASS |
| Microsoft | 2025 | 281,724 | 281,724 | PASS |
| Tesla | 2024 | 97,690 | 97,690 | PASS |
| Tesla | 2025 | 94,827 | 94,827 | PASS |
| Apple | 2023 | 383,285 | 383,285 | PASS |
| Apple | 2025 | 416,161 | 416,161 | PASS |

The chatbot correctly returned the revenue values for all tested
company and fiscal year combinations.